### PR TITLE
Allow to customize region name for a custom region with a user define…

### PR DESCRIPTION
**Pull request by fmassot**
…d endpoint.

### How was this PR tested?

Test added to check the region name set by an env variable.

### CODE REVIEW

**Comment by laurids-reichardt** [comment]
Would love to see this get merged. :)

File: quickwit-storage/src/object_storage/s3_compatible_storage.rs [hunk 1]
```diff
@@ -105,8 +105,17 @@ fn region_from_str(region_str: &str) -> anyhow::Result<Region> {
              http:// endpoint"
         );
     }
+
+    // For some storage provider like Cloudflare R2, we allow to to set the name of the region
+    // from `AWS_REGION` env variable.
+    let region_name = if let Ok(env_var) = std::env::var("AWS_REGION") {
```
Comment (guilload) [inline quickwit-storage/src/object_storage/s3_compatible_storage.rs:111]:
```rust
let region_name =
        std::env::var("AWS_REGION").unwrap_or_else(|_| "qw-custom-endpoint".to_string());
```

**Review by guilload — APPROVED** [review]

**Comment by fmassot** [comment]
@laurids-reichardt merged :)

**Comment by laurids-reichardt** [comment]
Awesome! Thanks! :)

**Pull request merged**